scripts/scraper.ts

```diff
@@ -0,0 +1,341 @@
+/* eslint-disable sonarjs/os-command */
+import { execSync } from "child_process";
+import { writeFileSync, mkdirSync, existsSync } from "fs";
+import { join } from "path";
+import crypto from "crypto";
+
+// Repository configuration
+interface RepoConfig {
+  owner: string;
+  repos: string[];
+}
+
+const repositories: RepoConfig[] = [
+  {
+    owner: "ubiquity-os-marketplace",
+    repos: [
+      "text-conversation-rewards",
+      "command-start-stop",
+      "text-vector-embeddings",
+      "command-wallet",
+      "daemon-pricing",
+      "daemon-disqualifier",
+      "command-query",
+      "command-ask",
+    ],
+  },
+  {
+    owner: "ubiquity-os",
+    repos: ["plugin-sdk", "ubiquity-os-plugin-installer", "ubiquity-os-kernel"],
+  },
+  {
+    owner: "ubiquity-os",
+    repos: ["pay.ubq.fi", "work.ubq.fi", "ubiquity-dollar"],
+  },
+];
+
+// Ensure data directory exists
+const dataDir = join(process.cwd(), "data");
+if (!existsSync(dataDir)) {
+  mkdirSync(dataDir);
+}
+
+const trainOutputPath = join(dataDir, "fine_tuning_train.jsonl");
+const validationOutputPath = join(dataDir, "fine_tuning_validation.jsonl");
+const TRAIN_SPLIT = 0.8; // 80% training, 20% validation
+
+// Valid time labels
+const VALID_TIME_LABELS = new Set([
+  "Time: <15 Minutes",
+  "Time: <1 Day",
+  "Time: <1 Week",
+  "Time: <1 Month",
+  "Time: <1 Hour",
+  "Time: <2 Hours",
+  "Time: <4 Hours",
+]);
```

Comment:
I'm not sure if this is the place I should leave the comment, but I think the model should be able to do more precise estimates. I only designed the course estimates to reduce cognitive overhead for humans to set them. But more precision should affect pricing as well.

Reply:
Typically, using discrete categories like Story Points or similar methods works better in agile environments. If we want to move towards continuous pricing estimates, we'd likely need to revise the current system. The main challenge is that the current data is based on discrete bins, which makes this shift difficult.

Reply:
I was checking out the training data and it looks like there are tasks in there that are not completed yet, just wondering if that would affect the training, I think so but I'm unsure. 

Also, it seems you used `issue.created` and `issue.closed` to detect completion time, but wouldn't the PR timestamps be more fit for purpose?

Curious as I've only done a small bit of training and I was wondering these things myself at the time.